packages/api/engine/openseatypes.go

```diff
@@ -0,0 +1,557 @@
+	BlockHash        string              `json:"block_hash"`
+	Timestamp        string              `json:"timestamp"`
+}
+
+// AssetBundle is the bundle of an asset
+type AssetBundle struct {
+	Maker         *Account       `json:"maker"`
+	Slug          string         `json:"slug"`
+	Assets        []*Asset       `json:"assets"`
+	Name          string         `json:"name"`
+	Description   string         `json:"description"`
+	ExternalLink  string         `json:"external_link"`
+	AssetContract *AssetContract `json:"asset_contract"`
+	Permalink     string         `json:"permalink"`
+	SellOrders    interface{}    `json:"sell_orders"`
+}
+
+// DevFeePaymentEvent is fee transfer event from OpenSea to Dev, It appears to be running in bulk on a regular basis.
+type DevFeePaymentEvent struct {
+	EventType      string       `json:"event_type"`
+	EventTimestamp string       `json:"event_timestamp"`
+	AuctionType    interface{}  `json:"auction_type"`
+	TotalPrice     interface{}  `json:"total_price"`
+	Transaction    Transaction  `json:"transaction"`
+	PaymentToken   PaymentToken `json:"payment_token"`
+}
+
+// EventType is the type of event
+type EventType string
+
+const (
+	// EventTypeNone is an undefined type of event
+	EventTypeNone EventType = ""
+
+	// EventTypeCreated is a creation event
+	EventTypeCreated EventType = "created"
+
+	// EventTypeSuccessful is a successful event
+	EventTypeSuccessful EventType = "successful"
+
+	// EventTypeCancelled is a canceled event
+	EventTypeCancelled EventType = "cancelled"
+
+	// EventTypeBidEntered is a bid event
+	EventTypeBidEntered EventType = "bid_entered"
+
+	// EventTypeBidWithdrawn is bid withdrawal event
+	EventTypeBidWithdrawn EventType = "bid_withdrawn"
+
+	// EventTypeTransfer is a transfer event
+	EventTypeTransfer EventType = "transfer"
+
+	// EventTypeApprove is an approval event
+	EventTypeApprove EventType = "approve"
+
+	// EventTypeCompositionCreated is a composition event
+	EventTypeCompositionCreated EventType = "composition_created"
+)
+
+// AuctionType is the type of an auction
+type AuctionType string
+
+const (
+	// AuctionTypeNone is an undefined type of auction
+	AuctionTypeNone AuctionType = ""
+
+	// AuctionTypeEnglish is an english auction
+	AuctionTypeEnglish AuctionType = "english"
+
+	// AuctionTypeDutch is a dutch auction
+	AuctionTypeDutch AuctionType = "dutch"
+
+	// AuctionTypeMinPrice is a min price auction
+	AuctionTypeMinPrice AuctionType = "min-price"
+)
+
+// Number is a big int
+type Number string
+
+// Big converts a string number to a big int
+func (n Number) Big() *big.Int {
+	s := strings.Split(string(n), ".")
+	r, _ := new(big.Int).SetString(s[0], 10)
+	return r
+}
+
+// Address is an address
+type Address string
+
+// NullAddress is the null address
+const NullAddress Address = "0x0000000000000000000000000000000000000000"
+
+// IsHexAddress returns true if a string is a hex address
+func IsHexAddress(s string) bool {
```

Comment:
a lot of these helper methods are available in go-ethereums `common` package

https://github.com/ethereum/go-ethereum/blob/master/common/types.go#L221